code/modules/food_and_drinks/restaurant/customers/_customer.dm

```diff
@@ -84,3 +84,20 @@
 		var/mutable_appearance/flag = mutable_appearance(customer.icon, "french_flag")
 		flag.appearance_flags = RESET_COLOR
 		return flag
+
+
+
+/datum/customer_data/japanese
+	nationality = "Space-Japanese"
+	prefix_file = "strings/names/japanese_prefix.txt"
+	base_icon = "japanese"
+	clothing_sets = list("japanese_animes")
+
+	found_seat_lines = list("Konnichiwa!", "Arigato gozaimasuuu~", "I hope there's some beef stroganoff...")
+	cant_find_seat_lines = list("I want to sit under the cherry tree already, senpai!", "Give me a seat before my Tsundere becomes Yandere!", "This place has less seating than a capsule hotel in space Tokyo!")
+	leave_mad_lines = list("I can't believe you did this! WAAAAAAAAAAAAAH!!", "I-It's not like I ever wanted your food! B-baka...")
+	leave_happy_lines = list("Oh CHEF! This is the happiest day of my life. I love you!", "I take a potato chip.... AND EAT IT!", "Itadakimasuuu~", "Gochisousama desu!")
+	wait_for_food_lines = list("No food yet? I guess it can't be helped.", "I can't wait to finally meet you burger-sama...", "Hurry up,")
```

Comment:
i'll say it; these tourist bot dialogs are starting to get a bit racist

Reply:
The Simpsons Theory: its not racist if you make fun of everyone

Reply:
I'll keep this review open just for bikeshed containment, in short I don't agree because when compared to every other customer stereotype they are exactly that but this one is honestly way more like a bunch of anime tropes/jokes crammed into a sprite. Hard to even tie it to something stereotypical of Japanese people in any way

Reply:
resolved over discord, actually